src/site/content/pages.njk

```diff
@@ -2,4 +2,15 @@
 noindex: true
 permalink: "/pages.json"
 ---
-{{ collections.pages | dump(2) | safe }}
+{% if site.env === 'prod' %}
+  {% set posts = collections.all | livePosts | algoliaIndexable %}
+{% else %}
+  {% set posts = [] %}
+{% endif %}
+
+[
+  {% for post in posts %}
+    {{ post | algoliaItem | stringify | safe }}
```

Comment:
There is in-built filter "dump". Maybe you can use it here instead of custom stringify?